### PR TITLE
Added uncrustify config

### PR DESCRIPTION
**Pull request by manuelmessner**
There are two problems with this config.
1. function pointer:
        `type (*ptr)(args)`
    will evaluate to
        `type (* ptr)(args)`
1. multi line function calls

```
        assignement = func(
            very_long_arg,
            very_long_arg
        );
```

```
will evaluate to
```

```
        assignement = func(
            very_long_arg,
            very_long_arg
                      );
```

After hours of searching, I didn't find a solution for these problems.
Feel free to search.

### CODE REVIEW

File: etc/uncrustify.cfg [hunk 1]
```diff
@@ -0,0 +1,1571 @@
...
+sp_inv                                   = remove   # ignore/add/remove/force
+
+# Add or remove space after the '&' (address-of) operator. Default=Remove
+# This does not affect the spacing after a '&' that is part of a type.
+sp_addr                                  = remove   # ignore/add/remove/force
+
+# Add or remove space around the '.' or '->' operators. Default=Remove
+sp_member                                = remove   # ignore/add/remove/force
+
+# Add or remove space after the '*' (dereference) operator. Default=Remove # This does not affect the spacing after a '*' that is part of a type.  sp_deref                                 = remove   # ignore/add/remove/force # Add or remove space after '+' or '-', as in 'x = -5' or 'y = +7'. Default=Remove sp_sign                                  = remove   # ignore/add/remove/force # Add or remove space before or after '++' and '--', as in '(--x)' or 'y++;'. Default=Remove
```
Comment (matthiasbeyer) [inline etc/uncrustify.cfg]: Please fix.

**Comment by TheNeikos** [comment]
What does the `indent_paren_nl` option do? Would it solve the second problem?

**Comment by TheNeikos** [comment]
And does `sp_after_ptr_star_func` setting it to `remove` fix the first problem?

**Comment by matthiasbeyer** [comment]
Targets #26 , btw.

**Comment by manuelmessner** [comment]
@TheNeikos 
Unfortunately neither your first option nor the second one fixes the existing problems.

**Comment by matthiasbeyer** [comment]
[Related reddit](http://www.reddit.com/r/C_Programming/comments/2hx8oj/styleformatting_with_uncrustify_some_problems_are/) post. No answers yet.

**Comment by TheNeikos** [comment]
I don't think this can be done anytime soon. Should we put that on ignore? Is that possible?

**Comment by neithernut** [comment]
We can label it "on hold".

**Comment by matthiasbeyer** [comment]
I guess we will not finish this one. But I guess our style is pretty good, so I propose: Close this unmerged.

**Comment by manuelmessner** [comment]
@matthiasbeyer 
ACK. I really don't know what to change ...

**Pull request closed without merging**